protocol/xmrmaker/claim.go

```diff
@@ -135,41 +191,63 @@ func claimRelayer(
 		return ethcommon.Hash{}, err
 	}
 
-	abi, err := abi.JSON(strings.NewReader(contracts.SwapFactoryMetaData.ABI))
+	calldata, err := getClaimTxCalldata(relayerCommission, contractSwap, secret)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	feeValue, err := calculateRelayerCommission(contractSwap.Value, relayerCommission)
+	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	calldata, err := abi.Pack("claimRelayer", *contractSwap, secret, feeValue)
+	err = waitAndCheck(ctx, ec, txHash)
 	if err != nil {
 		return ethcommon.Hash{}, err
 	}
 
-	txHash, err := rc.SubmitTransaction(contractAddr, calldata)
-	if err != nil {
-		return ethcommon.Hash{}, err
-	}
+	return txHash, nil
+}
 
+func waitAndCheck(ctx context.Context, ec *ethclient.Client, txHash ethcommon.Hash) error {
 	// wait for inclusion
 	receipt, err := block.WaitForReceipt(ctx, ec, txHash)
 	if err != nil {
-		return ethcommon.Hash{}, err
+		return err
 	}
 
 	if receipt.Status == 0 {
-		return ethcommon.Hash{}, fmt.Errorf("transaction failed")
+		return fmt.Errorf("transaction failed")
```

Comment:
`WaitForReceipt` above already did this status check: https://github.com/AthanorLabs/atomic-swap/blob/master/ethereum/block/wait_for_receipt.go#L44